benchmark worker updates

vtjnash opened the pull request:
I knew about the sudo-interpolation-ENV thing, I just have not been good about making PRs for my branches

vtjnash committed:
ensure !has_nondefault_cmd_flags

vtjnash committed:
update all deps

vtjnash committed:
corrected path for bare repository

vtjnash committed:
fix typo in sudo

maleadt commented on src/build.jl:
```diff
@@ -55,7 +55,7 @@ function build_julia!(config::Config, build::BuildRef, logpath, prnumber::Union{
     mkpath(dirname(mirrordir), mode=0o755)
     mkpidlock(mirrordir * ".lock") do
-        if ispath(mirrordir, ".git")
+        if ispath(mirrordir, "config")
```
Isn't this a git clone?

vtjnash replied:
no, it is a mirror

vtjnash commented:
now failing with ```
│ UndefVarError: benchresults not defined
│ Stacktrace:
│  [1] execute_benchmarks!
│    @ /nanosoldier/Nanosoldier.jl/src/jobs/BenchmarkJob.jl:467
```

vtjnash commented on src/jobs/BenchmarkJob.jl:
```diff
@@ -340,7 +340,7 @@ function execute_benchmarks!(job::BenchmarkJob, juliapath, whichbuild::Symbol)
     end
 
-    run(sudo(cfg.user, setenv(`$(setenv(juliacmd, nothing)) -e 'using Pkg; Pkg.instantiate(); Pkg.status()'`; dir=builddir)))
+    run(sudo(cfg.user, `$(setenv(juliacmd, nothing, dir=builddir)) -e 'using Pkg; Pkg.instantiate(); Pkg.status()'`))
```
it gets moved to `sudo` either way

vtjnash committed:
fix variable from #121

maleadt commented:
Yeah sorry for the typo's. It's too bad that it's near-impossible to test the benchmark bot, since the code and configuration here is so specific to your set-up (e.g., `cset`, specific users, etc).

vtjnash commented:
There is a script to activate a machine as a worker (only do this inside docker, or a machine you actually want to convert though). But not to worry either, as these were pretty simple, and I just run a few trial of nanosoldier whenever I go to update it to check

vtjnash committed:
switch xz to zstd

vtjnash committed:
fixes and clarifications for provision scripts

vtjnash committed:
move run_base_ci to run in tmux as nanosoldier

This makes it easier for anyone with access to log in and view or change the live status.

vtjnash commented:
Oof, that looks like a lot of conflicts. I really need a successor to take over updating this machine
